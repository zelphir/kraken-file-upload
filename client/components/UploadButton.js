import React from 'react'
import PropTypes from 'prop-types'
import { Icon } from '@blueprintjs/core'

const UploadButton = ({ large, onUpload }) => {
  return (
    <React.Fragment>
      <input type="file" name="file" id="file" className="inputfile" onChange={onUpload} />
      <label htmlFor="file" className={`pt-button pt-intent-primary${large ? ' pt-large' : ''}`}>
        <Icon icon="upload" />
        <span>Upload a new file!</span>
      </label>
    </React.Fragment>
  )
}

UploadButton.propTypes = {
  large: PropTypes.bool,
  onUpload: PropTypes.func.isRequired
}

export default UploadButton
