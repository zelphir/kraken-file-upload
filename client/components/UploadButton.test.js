import React from 'react'
import { shallow } from 'enzyme'
import UploadButton from './UploadButton'

function setup(nextProps = {}) {
  const props = {
    large: false,
    onUpload: jest.fn(),
    ...nextProps
  }
  const wrapper = shallow(<UploadButton {...props} />)

  return {
    props,
    wrapper
  }
}

describe('<UploadButton />', () => {
  it('should match the snapshot', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the large version if large', () => {
    const { wrapper } = setup({ large: true })
    expect(wrapper.find('label').hasClass('pt-large')).toBe(true)
  })
})
