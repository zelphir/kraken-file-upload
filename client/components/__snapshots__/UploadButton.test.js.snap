// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadButton /> should match the snapshot 1`] = `
<React.Fragment>
  <input
    className="inputfile"
    id="file"
    name="file"
    onChange={[MockFunction]}
    type="file"
  />
  <label
    className="pt-button pt-intent-primary"
    htmlFor="file"
  >
    <Blueprint2.Icon
      icon="upload"
    />
    <span>
      Upload a new file!
    </span>
  </label>
</React.Fragment>
`;
